File: src/main/java/com/braggbnb109/service/PropertyTypeService.java
package com.braggbnb109.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb109.domain.PropertyType;
import com.braggbnb109.dto.PropertyTypeDTO;
import com.braggbnb109.dto.PropertyTypeSearchDTO;
import com.braggbnb109.dto.PropertyTypePageDTO;
import com.braggbnb109.dto.PropertyTypeConvertCriteriaDTO;
import com.braggbnb109.service.GenericService;
import com.braggbnb109.dto.common.RequestDTO;
import com.braggbnb109.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PropertyTypeService extends GenericService<PropertyType, Integer> {

	List<PropertyType> findAll();

	ResultDTO addPropertyType(PropertyTypeDTO propertyTypeDTO, RequestDTO requestDTO);

	ResultDTO updatePropertyType(PropertyTypeDTO propertyTypeDTO, RequestDTO requestDTO);

    Page<PropertyType> getAllPropertyTypes(Pageable pageable);

    Page<PropertyType> getAllPropertyTypes(Specification<PropertyType> spec, Pageable pageable);

	ResponseEntity<PropertyTypePageDTO> getPropertyTypes(PropertyTypeSearchDTO propertyTypeSearchDTO);
	
	List<PropertyTypeDTO> convertPropertyTypesToPropertyTypeDTOs(List<PropertyType> propertyTypes, PropertyTypeConvertCriteriaDTO convertCriteria);

	PropertyTypeDTO getPropertyTypeDTOById(Integer propertyTypeId);







}

File: braggbnb109_ui/src/services/PropertyTypeService.js
import http from "../http-common"; 

class PropertyTypeService {
  getAllPropertyTypes(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/propertyType/propertyTypes`, searchDTO);
  }

  get(propertyTypeId) {
    return this.getRequest(`/propertyType/${propertyTypeId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/propertyType?field=${matchData}`, null);
  }

  addPropertyType(data) {
    return http.post("/propertyType/addPropertyType", data);
  }

  update(data) {
  	return http.post("/propertyType/updatePropertyType", data);
  }
  
  uploadImage(data,propertyTypeId) {
  	return http.postForm("/propertyType/uploadImage/"+propertyTypeId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PropertyTypeService();
